partitura/utils/music.py

```diff
@@ -3,15 +3,27 @@
 """
 This module contains music related utilities
 """
+from __future__ import annotations
+
 from collections import defaultdict
 import re
 import warnings
 import numpy as np
 from scipy.interpolate import interp1d
 from scipy.sparse import csc_matrix
-from typing import Union, Callable
+from typing import Union, Callable, Optional, TYPE_CHECKING
 from partitura.utils.generic import find_nearest, search, iter_current_next
 
+from partitura.utils.misc import deprecated_alias
+
+if TYPE_CHECKING:
```

Comment:
Hello, I raise the minor issue but the following lines are not very clear to me:

```python
if TYPE_CHECKING:
    # Import typing info for typing annotations.
    # For this to work we need to import annotations from __future__
    # Solution from
    # https://medium.com/quick-code/python-type-hinting-eliminating-importerror-due-to-circular-imports-265dfb0580f8
    from partitura.score import ScoreLike
    from partitura.performance import PerformanceLike, Performance
    ```
Why are these optional imports?

Reply:
These imports are necessary for typing annotations (since the type of many input variables in the utilities that are `ScoreLike` or `PerformanceLike`, which would create circular imports.